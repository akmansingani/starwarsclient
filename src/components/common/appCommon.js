import React, { Fragment } from "react";
import { element } from "prop-types";

export const MovieTitle = ({ varTitle }) => (
  <Fragment>
    {varTitle === "" ? (
      ""
    ) : (
      <div>
        <div className="quesText mb-4">
          Which of all StarWars movies has longest opening crawl ?
        </div>
        <div className="ansText mb-4">{varTitle}</div>
      </div>
    )}
  </Fragment>
);

export const MostCharacter = ({ varCharacter }) => (
  <Fragment>
    {varCharacter === "" ? (
      ""
    ) : (
      <div>
        <div className="quesText mb-4">
          What character(person) appeared in the most of StarWar Films ?
        </div>
        <div className="ansText mb-4">{varCharacter}</div>
      </div>
    )}
  </Fragment>
);

export const MostSpecies = ({ varSpecies }) => (
         <Fragment>
           {varSpecies.length === 0 ? (
             ""
           ) : (
             <div>
               <div className="quesText mb-4">
                 What species appeared in the most number of StarWars films (Top 5) ?
               </div>
               {varSpecies.map((item, index) => (
                 <div key={index} className="ansText mb-4">
                   {item.speciesname} ({item.speciescount})
                 </div>
               ))}
             </div>
           )}
         </Fragment>
       );

       export const MostPlanet = ({ varPlanets }) => (
                <Fragment>
                  {varPlanets.length === 0 ? (
                    ""
                  ) : (
                    <div>
                      <div className="quesText mb-4">
                        What planet in Star Wars universe provided largest
                        number of vehicle pilots (Top 5) ?
                      </div>
                      {varPlanets.map((item, index) => (
                        <div key={index} className="ansText mb-4">
                          Planet : {item.planetname} - Pilots : (
                          {item.people_count}) - 
                          {item.people_list.map((element,i) => (
                           <span key={i}> {element.people_name} 
                           {i !== item.people_list.length -1 ? " , " : ""} 
                           </span>
                          ))}
                        </div>
                      ))}
                    </div>
                  )}
                </Fragment>
              );