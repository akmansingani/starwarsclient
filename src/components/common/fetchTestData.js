import axios from "axios";

export const API = process.env.REACT_APP_API_URL;
export const fetchData = async endpoint => {
         const url = `${API}/endpoint`;
         return await axios.get(url);
};

