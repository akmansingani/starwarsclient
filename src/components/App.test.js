import React from "react";
import renderer from "react-test-renderer";
import { MovieTitle, MostCharacter,MostSpecies } from "./common/appCommon";

import axios from "axios";
import { fetchData } from "./common/fetchTestData";

jest.mock("axios");

describe("Start Test", () => {
  it("Init Test Case", () => {
    expect(true).toEqual(true);
  });
});


describe("MovieTitle", () => {
  test("snapshot renders", () => {
    const component = renderer.create(<MovieTitle varTitle={"test"} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("fetches successfully data from an API", async () => {
    const data = { status: "success", response: "A New Hope" };
    axios.get.mockImplementationOnce(() => Promise.resolve(data));
    await expect(fetchData("home/getTitleOpenCrawl")).resolves.toEqual(data);
  });

});

describe("MostAppearedCharacter", () => {
  
  test("snapshot renders", () => {
    const component = renderer.create(<MostCharacter varCharacter={"test"} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  
  it("fetches successfully data from an API", async () => {
      const data = { status: "success", response: "A New Hope" };
      axios.get.mockImplementationOnce(() => Promise.resolve(data));
      await expect(fetchData("home/getCharacterMostAppeared")).resolves.toEqual(
        data
      );
  });


});

describe("MostNumberOfSpecies", () => {

  var resp = [
    {
      speciesname: "Human",
      speciescount: 4
    }
  ];

  test("snapshot renders", () => {
    const component = renderer.create(<MostSpecies varSpecies={resp} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("fetches successfully data from an API", async () => {
    const data = { status: "success", response: resp };
    axios.get.mockImplementationOnce(() => Promise.resolve(data));
    await expect(fetchData("home/getMostNumberOfSpecies")).resolves.toEqual(
      data
    );
  });
});




