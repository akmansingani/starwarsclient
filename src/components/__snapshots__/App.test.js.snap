// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MostAppearedCharacter snapshot renders 1`] = `
<div>
  <div
    className="quesText mb-4"
  >
    What character(person) appeared in the most of StarWar Films ?
  </div>
  <div
    className="ansText mb-4"
  >
    test
  </div>
</div>
`;

exports[`MostNumberOfPilots snapshot renders 1`] = `
<div>
  <div
    className="quesText mb-4"
  >
    What planet in Star Wars universe provided largest number of vehicle pilots (Top 5) ?
  </div>
  <div
    className="ansText mb-4"
  >
    Planet : 
    Tatooine
     - Pilots : (
    19
    ) -
    <span>
       
      Chewbacca - Wookie
       , 
    </span>
    <span>
       
      Nien Nunb - Sullustan
      
    </span>
  </div>
</div>
`;

exports[`MostNumberOfSpecies snapshot renders 1`] = `
<div>
  <div
    className="quesText mb-4"
  >
    What species appeared in the most number of StarWars films (Top 5) ?
  </div>
  <div
    className="ansText mb-4"
  >
    Human
     (
    4
    )
  </div>
</div>
`;

exports[`MovieTitle snapshot renders 1`] = `
<div>
  <div
    className="quesText mb-4"
  >
    Which of all StarWars movies has longest opening crawl ?
  </div>
  <div
    className="ansText mb-4"
  >
    test
  </div>
</div>
`;
